tests-js/loan-token/TradingwRBTCLoan.test.js

```diff
@@ -0,0 +1,147 @@
+const { BN } = require("@openzeppelin/test-helpers");
+
+const {
+	margin_trading_sending_loan_tokens,
+	margin_trading_sov_reward_payment,
+	margin_trading_sending_collateral_tokens,
+	margin_trading_sending_collateral_tokens_sov_reward_payment,
+	close_complete_margin_trade,
+} = require("./tradingFunctions");
+
+const FeesEvents = artifacts.require("FeesEvents");
+
+const {
+	getSUSD,
+	getRBTC,
+	getWRBTC,
+	getBZRX,
+	getSOV,
+	getLoanTokenLogic,
+	getLoanTokenLogicWrbtc,
+	getLoanToken,
+	getLoanTokenWRBTC,
+	loan_pool_setup,
+	set_demand_curve,
+	lend_to_pool,
+	getPriceFeeds,
+	getSovryn,
+	open_margin_trade_position_iBTC,
+} = require("../Utils/initializer.js");
+
+const wei = web3.utils.toWei;
+
+const oneEth = new BN(wei("1", "ether"));
+
+contract("LoanTokenTrading", (accounts) => {
+	let owner;
+	let sovryn, SUSD, WRBTC, RBTC, BZRX, loanToken, loanTokenWRBTC, SOV, priceFeeds;
+
+	before(async () => {
+		[owner] = accounts;
+	});
+
+	beforeEach(async () => {
+		SUSD = await getSUSD();
+		RBTC = await getRBTC();
+		WRBTC = await getWRBTC();
+		BZRX = await getBZRX();
+		priceFeeds = await getPriceFeeds(WRBTC, SUSD, RBTC, sovryn, BZRX);
+
+		sovryn = await getSovryn(WRBTC, SUSD, RBTC, priceFeeds);
+
+		const loanTokenLogicStandard = await getLoanTokenLogic();
+		const LoanTokenLogicWrbtc = await getLoanTokenLogicWrbtc();
+		loanToken = await getLoanToken(loanTokenLogicStandard, owner, sovryn, WRBTC, SUSD);
+		loanTokenWRBTC = await getLoanTokenWRBTC(LoanTokenLogicWrbtc, owner, sovryn, WRBTC, SUSD);
+		await loan_pool_setup(sovryn, owner, RBTC, WRBTC, SUSD, loanToken, loanTokenWRBTC);
+
+		SOV = await getSOV(sovryn, priceFeeds, SUSD);
+	});
+
+	describe("test the loan token trading logic with SUSD test token as collateral token and the wBTC as underlying loan token. ", () => {
+		/*
+      tests margin trading sending loan tokens.
+			process is handled by the shared function margin_trading_sending_loan_tokens
+			1. approve the transfer
+			2. send the margin trade tx
+			3. verify the trade event and balances are correct
+			4. retrieve the loan from the smart contract and make sure all values are set as expected
+    */
+		it("Test margin trading sending loan tokens", async () => {
+			await margin_trading_sending_loan_tokens(accounts, sovryn, loanTokenWRBTC, WRBTC, SUSD, priceFeeds, false);
+			await margin_trading_sov_reward_payment(accounts, loanTokenWRBTC, WRBTC, SUSD, SOV, FeesEvents);
+		});
+		// it("Test margin trading sending collateral tokens", async () => {
```

Comment:
WIP?

Reply:
commented tests which are not working correctly.
Hope to get help on this.

Reply:
the comment can be marked resolved only when it is resolved

Reply:
fixed by priceFeeds